### PR TITLE
Stripe connect account & charge definition updates

### PR DESCRIPTION
**Pull request by sarcastic-coder**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
* [Create Charge Docs](https://stripe.com/docs/api/charges/create)
* [Create Connect Account Docs](https://stripe.com/docs/api/accounts)
* [API Changelog](https://stripe.com/docs/upgrades)
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by jessetrinity** [comment]
@wjohnsto @CodeAnimal @sampsonjoliver @LinusU @brannon @kkamperschroer @starhoshi @bruun @galtalmor @htunnicliff @squirly @tzarger @ifiokjr @SimonSchick @yultyyev @cpsoinos @adamduren @saranshkataria @0xJoKe @delianides @gokulchandra

Any interest in this PR?

**Comment by sandersn** [comment]
@coffee-powered-coder this seems like a big change. Is this still for 6.25, or is it for a newer version? 

Paging @wjohnsto @CodeAnimal @sampsonjoliver @LinusU @brannon @kkamperschroer @starhoshi @bruun @galtalmor @htunnicliff @squirly @tzarger @ifiokjr @SimonSchick @yultyyev @cpsoinos @adamduren @saranshkataria @0xJoKe @delianides @gokulchandra

Mind taking a look? There's a lot of new stuff here that was presumably missing before. stripe is just at our threshold for "too popular to change without owner signoff", so I don't want to merge it with no knowledge of the package.

File: types/stripe/index.d.ts
Comment (viralpickaxe) [inline types/stripe/index.d.ts]: There is a new `IAddress`, please could you use this

**Review by viralpickaxe — CHANGES_REQUESTED** [review]
Thanks for your contribution! I added an `IAddress` interface recently, would you mind updating your changes to use it to reduce duplication plz :)

File: types/stripe/index.d.ts
Comment (coffee-powered-coder) [reply types/stripe/index.d.ts]: :+1: Implemented

**Comment by coffee-powered-coder** [comment]
@sandersn Bumped the package version as you suggested since there are properties deleted/moved elsewhere to correspond with the API.

File: types/stripe/index.d.ts
Comment (SimonSchick) [inline types/stripe/index.d.ts]: Could you kindly remove me as a contributor in this PR?
Getting waaay too many emails from this.

File: types/stripe/index.d.ts
Comment (viralpickaxe) [inline types/stripe/index.d.ts]: It should also be noted that these properties were moved rather than just deleted and therefore should be added back in their new place. But fine leaving that out of this PR

**Review by viralpickaxe — CHANGES_REQUESTED** [review]
I took a more thorough look through the changes against the API reference and noticed these

File: types/stripe/index.d.ts
Comment (coffee-powered-coder) [reply types/stripe/index.d.ts]: `support_email` and other properties already exist in the `business_profile` object at the top of the `IAccountShared` interface [here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/HEAD/types/stripe/index.d.ts#L283) in line with the API.

Comment (viralpickaxe) [reply types/stripe/index.d.ts]: Hrmm okay, but others such as `settings.branding.primary_color/logo/icon` etc. haven't been re-added

File: types/stripe/index.d.ts
Comment (viralpickaxe) [inline types/stripe/index.d.ts]: much clean. much wow. 😍

File: types/stripe/index.d.ts
Comment (viralpickaxe) [inline types/stripe/index.d.ts]: As mentioned in my previous comment, these `x_provided` properties can't be added to the `IAccountShared` interface as `IAccountUpdateOptions` and `IAccountCreationOptions` extend it and they aren't valid props on that create method.

`x_provided` props should be added to the `IAccount` interface and `x` props should be added to the `IAccountUpdateOptions` interface.

**Review by viralpickaxe — CHANGES_REQUESTED** [review]

File: types/stripe/index.d.ts
Comment (coffee-powered-coder) [reply types/stripe/index.d.ts]: They're all still under `settings.branding` of the `IAccountShared` object in [master](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/HEAD/types/stripe/index.d.ts#L376) and haven't moved in my PR. Looks like there was [another PR merged](https://github.com/DefinitelyTyped/DefinitelyTyped/commit/a408d42b1add1b32a21c9454eab6b529112e9d91) containing half of the `IAccountShared` properties and not clearing up the old ones.

File: types/stripe/index.d.ts
Comment (SimonSchick) [inline types/stripe/index.d.ts]: ❤️

**Review by SimonSchick — APPROVED** [review]

**Comment by kbala** [comment]
Time to merge this PR?

**Comment by kbala** [comment]
@typescript-bot Time to merge this PR?

**Review by kbala-ami — APPROVED** [review]
As per latest Stripe documentation, these changes are updated.

**Comment by weswigham** [comment]
@viralpickaxe you need to approve/dismiss your reviews ❤️

**Comment by kbala** [comment]
@coffee-powered-coder can you please answer the @viralpickaxe reviews?

**Review by viralpickaxe — APPROVED** [review]

**Comment by kbala** [comment]
@viralpickaxe @coffee-powered-coder @typescript-bot expecting to merge soon 👍

**Comment by kbala** [comment]
@coffee-powered-coder expecting to merge....

**Comment by kbala** [comment]
@typescript-bot Yep, its up. Thanks!

**Pull request merged**
